scio-core/src/main/scala/com/spotify/scio/io/dynamic/package.scala

```diff
@@ -122,6 +129,34 @@ package object dynamic {
 
   }
 
+  implicit class DynamicProtobufSCollection[T <: Message](private val self: SCollection[T])
+      extends AnyVal {
+
+    def saveAsDynamicProtobufFile(
+      path: String,
+      numShards: Int = 0,
+      suffix: String = ".protobuf",
+      codec: CodecFactory = CodecFactory.deflateCodec(6),
+      metadata: Map[String, AnyRef] = Map.empty
+    )(destinationFn: T => String)(implicit cd: ClassTag[T], coder: ScioCoder[T]): Future[Tap[T]] = {
+      val elemCoder = CoderMaterializer.beam(
+        self.context,
+        coder
```

Comment:
we know already that this is a proto so we don;t need to infer the coder. We can:

```scala
val protoCoder =
  Coder.protoMessageCoder[Message](ct.asInstanceOf[ClassTag[Message]]).asInstanceOf[Coder[T]]
val elemCoder = CoderMaterializer.beam(self.context, protoCoder)
```